### DISCUSSION TITLE
Supplied data: an autogenerated set for `tags` names

### DISCUSSION THREAD

**Discussion by pdehaan** [11ty/eleventy#4031] [Enhancement Queue]
**Is your feature request related to a problem? Please describe.**
Seems to be a lot of confusion around the best way to get a list of tags (for generating archive pages or whatever).
Not sure if there'd be any way of automatically creating a "collections.tags" collection for any of the `tags` that somebody specifies in a template.

I think there are way of using `Object.keys(collections)` via a filter, but even that has some flaws.

**Describe the solution you'd like**
A custom "tags" collection which 11ty manages and dedupes based on whatever "tags" somebody specifies in their templates. Not sure if it should just possibly be an array of strings or if it'd need to be something more complex (a map of strings and associated pages, but that might be overkill).

**Describe alternatives you've considered**
Managing the list myself, but using `Object.keys(collections)` can include manually added collections, so then you have to do a bunch of filtering to keep out unwanted collection names from your tag list.

**Comment by zachleat** [comment]
> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!

**Comment by zachleat** [comment]
Sorry for the late response!

There are three different types of collection name sets: `tags`, configuration API collection names, and the _special_ autogenerated one for `all`. I could see the benefit of having a tags set, I’m curious if there would also be benefit in the configuration API collection names set too.
